### PR TITLE
[C#] nested-types new exercise

### PR DESCRIPTION
**Pull request by mikedamay**
Plese review [_NestedTypes.cs_](https://github.com/mikedamay/v3/blob/csharp/nested-types/languages/csharp/exercises/concept/nested-types/NestedTypes.cs) - starting point, [_Example.cs_](https://github.com/mikedamay/v3/blob/csharp/nested-types/languages/csharp/exercises/concept/nested-types/.meta/Example.cs) - refactored result, [_instructions.md_](https://github.com/mikedamay/v3/blob/csharp/nested-types/languages/csharp/exercises/concept/nested-types/.docs/instructions.md), [_design.md_](https://github.com/mikedamay/v3/blob/csharp/nested-types/languages/csharp/exercises/concept/nested-types/.meta/design.md) and [_NestedTypesTests.cs_](https://github.com/mikedamay/v3/blob/csharp/nested-types/languages/csharp/exercises/concept/nested-types/NestedTypesTests.cs).

### CODE REVIEW

**Comment by ErikSchierboom** [comment]
I like this! Very nice!

**Comment by mikedamay** [comment]
@ErikSchierboom while you have that in mind I was thinking for `namespaces` we might talk about rival remote control car brands needing to have their own version of telemetry, motor, chassis etc. so there might be a RedCar namespace and a BlueCar namespace with the same named classes but unrelated implementations and APIs.  It doesn't make a lot of sense but none of it does really.  I was hoping someone would pick up `namespaces` today.

**Comment by ErikSchierboom** [comment]
@mikedamay That could work. Maybe as a small tweak they could be two divisions in the same company trying to build the same car but using different methodologies/techniques/materials or something like that?

**Comment by mikedamay** [comment]
That's almost realistic!

**Comment by ErikSchierboom** [comment]
> That's almost realistic!

Ha 🤣

**Comment by mikedamay** [comment]
closed to finalize - will reopen shortly

**Comment by mikedamay** [comment]
All files complete and proofed except _csharp/config.json_.

File: languages/csharp/exercises/concept/nested-types/.docs/hints.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+## General
+
+This documentation of [nested types][nested-types] details the syntax.
+
+## 1. Separate concerns between the car itself and the telemetry system
+
+Members of `RemoteControlCar` suffixed with "\_Telemetry" could be moved to a separate class.
```
Comment (ErikSchierboom) [inline languages/csharp/exercises/concept/nested-types/.docs/hints.md:7]:
```suggestion
- Members of `RemoteControlCar` suffixed with "\_Telemetry" could be moved to a separate class.
```

File: languages/csharp/exercises/concept/nested-types/.docs/hints.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+## General
+
+This documentation of [nested types][nested-types] details the syntax.
```
Comment (ErikSchierboom) [inline languages/csharp/exercises/concept/nested-types/.docs/hints.md:3]:
```suggestion
- This documentation of [nested types][nested-types] details the syntax.
```

File: languages/csharp/exercises/concept/nested-types/.docs/hints.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+## 1. Separate concerns between the car itself and the telemetry system
+
+Members of `RemoteControlCar` suffixed with "\_Telemetry" could be moved to a separate class.
+
+## 2. Prevent other code taking too many dependencies on the Telemetry type
+
+[Interfaces][interfaces] are the secret sauce here.
```
Comment (ErikSchierboom) [inline languages/csharp/exercises/concept/nested-types/.docs/hints.md:11]:
```suggestion
- [Interfaces][interfaces] are the secret sauce here.
```

File: languages/csharp/exercises/concept/nested-types/.docs/hints.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+## 3. Prevent other code from taking dependencies on the Speed struct
+
+Give some consideration to [access levels][access-levels].
```
Comment (ErikSchierboom) [inline languages/csharp/exercises/concept/nested-types/.docs/hints.md:15]:
```suggestion
- Give some consideration to [access levels][access-levels].
```

File: languages/csharp/exercises/concept/nested-types/NestedTypes.cs [hunk 5]
```diff
@@ -0,0 +1,77 @@
+public class RemoteControlCar
+{
+    public string CurrentSponsor { get; private set; }
+
+    private Speed currentSpeed;
+
+    public void Calibrate_Telementry()
+    {
+
```
[the remark this answers is not in the document]
Comment (mikedamay) [reply languages/csharp/exercises/concept/nested-types/NestedTypes.cs:11]: Done

**Pull request merged**